src/main/java/lambdastreams/Person.java

```diff
@@ -0,0 +1,87 @@
+package lambdastreams;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Person {
+    static final int ADULT_FROM_AGE = 18;
+    private int age;
+    private String firstName;
+    private String lastName;
+
+    public Person(String firstName, String lastName, int age) {
+
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
+    }
+
+    public static void main(String[] args) {
+
+        System.out.println("List of persons:");
+        List<Person> personList = new ArrayList<>();
+        personList.add(new Person("Peter", "Patterson", 21));
+        personList.add(new Person("Paul", "Walker", 31));
+        personList.add(new Person("Steve", "Runner", 17));
+        personList.add(new Person("Arnold", "", -1));
+        personList.add(new Person(" ", "Stevenson", 19));
+        personList.add(new Person("   ", "Stevenson", 19));
+        personList.add(new Person("      ", "Stevenson", 19));
+        personList.add(new Person(" Arnold", "Stevenson", 19));
+        personList.add(null);
+        personList.add(new Person("Aaron", "Bortnicker", 18));
+
+        for (Person person : personList) {
+            System.out.println(person);
+        }
+
+        System.out.println("\nList of unique first names of adult persons:\n" + findFirstNamesOfAdults(personList));
+
+    }
+
+    public static Set<String> findFirstNamesOfAdults(Collection<Person> persons) {
+
+        Set<String> personsFirstNames;
+        personsFirstNames = persons.stream()
+                .filter(Objects::nonNull)
+                .filter(adult -> adult.age >= ADULT_FROM_AGE)
```

Comment:
please use get() methods instead of variables so this method can be used outside this class

Reply:
done.